### PR TITLE
Clarified code of GOMidiEventRecvTab

### PR DESCRIPTION
**Pull request by oleg68**
This is the first PR related to #1392.

1. Added some comments in the GOMidiEventRecvTab header
2. Introduced the new variable GOMidiEventRecvTab::m_ReceiverType for eliminating calling m_midi.GetType()
3. Removed extra returns from GOMidiEventRecvTab::DetectEvent()

It is just refactoring. No GO behavior should be changes.

### CODE REVIEW

**Comment by oleg68** [comment]
@rousseldenis @larspalo could you approve this PR because it delays completing #1392

File: src/grandorgue/dialogs/midi-event/GOMidiEventRecvTab.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ class GOMidiEventRecvTab : public wxPanel,
 
   GOMidiReceiverBase *m_original;
   GOMidiReceiverEventPatternList m_midi;
+  GOMidiReceiverType m_ReceiverType;
```
Comment (larspalo) [inline src/grandorgue/dialogs/midi-event/GOMidiEventRecvTab.h:41]: I'm somewhat concerned about this. I understand the supposed "simplification", but GOMidiReceiverType is a typedef:ed enum that's buried in another header file (src/core/midi/GOMidiReceiverEventPatternList.h) that all breath of old c style coding that should be structured in a better way, in my opinion. To me, this "clarification" is not really so "clear" as there is no class or separate file that can be easily search for to find the defined type.

Comment (oleg68) [reply src/grandorgue/dialogs/midi-event/GOMidiEventRecvTab.h:41]: You are right. But extracting GOMidiReceiverType is a subject of a separate pr #1664.

**Comment by oleg68** [comment]
@larspalo @rousseldenis could you approve this PS because it blocks #1392 ?

**Review by rousseldenis — APPROVED** [review]

**Review by larspalo — APPROVED** [review]

**Pull request merged**